h/views/api/bulk.py

```diff
@@ -30,13 +32,60 @@ def from_byte_stream(cls, byte_stream, executor, observer=None):
             return None
 
 
+class AuthorityCheckingExecutor(AutomaticReportExecutor):
```

Comment:
This won't live here in the end, but it'll do for now

Reply:
The inheritance here makes me uncomfortable. I understand inheritance to be a tightly coupled "is a" relationship, and I don't think an `AuthorityCheckingExecutor` "is an" `AutomaticReportExecutor`.

`AutomaticReportExecutor` doesn't have any methods other than `execute_batch()` and, even if it grows more methods in future, `AuthorityCheckingExecutor` doesn't call them (doesn't call `super()` for anything other than `execute_batch()`).

It looks like what's actually going on here is that `AuthorityCheckingExecutor` is a standalone executor that makes use of an `AutomaticReportExecutor` for reporting. So I think composition might be a more appropriate pattern:

```python
class AuthorityCheckingExecutor(Executor):

    def __init__(self, authority="lms.hypothes.is"):
        self._automatic_report_executor = AutomaticReportExecutor()
        ...

    def configure(self, config):
        ...

    def execute_batch(self, command_type, data_type, default_config, batch):
        ...

        self._automatic_report_executor.execute_batch(command_type, data_type, default_config, batch)

    ...
```

I also noticed that `AutomaticReportExecutor`'s one method (`execute_batch()`) doesn't use `self`, so the class could perhaps be turned into a `create_fake_report()` function that `AuthorityCheckingExecutor` could call:

```python
class AuthorityCheckingExecutor(Executor):

    ...

    def execute_batch(self, command_type, data_type, default_config, batch):
        ...

        create_fake_report(command_type, data_type, default_config, batch)

    ...
```

Reply:
It won't matter in the end as the `AutomaticReportingExecutor` is only while we are faking reports, so pretty soon this won't inherit from this.